proto/kuksa/val/v2/val.proto

```diff
@@ -280,7 +282,12 @@ message BatchActuateStreamRequest {
   repeated ActuateRequest actuate_requests = 1;
 }
 
+// Message that can be used by provider to indicate that an actuation request was not accepted.
+// It is optional for the provider to use this message, and for performance reasons it is recommended
+// not to send it upon success.
 message BatchActuateStreamResponse {
+  SignalID signal_id   = 1;
+  Error error = 2;
```

Comment:
This is a bit of a architectural question, but do we really want a single `Error` enum that will contain every error possible for all functions in the whole service?

Wouldn't it be better to have specific errors for different functions, i.e. an `ActuationError` that only contains the errors possible when trying to actuate?

Reply:
I think the enum I used matched quite well. At least I can find use-cases for all of them

* OK - Good to have a default value that indicates no error
* INVALID_ARGUMENT - value is received is out of expected range
* NOT_FOUND - signal id is not known by provider
* PERMISSION_DENIED - signal is known by provider, but it does not accept that it is changes. A possible example could be that the provider has not registered itself for the signal id, so it does not expect to get it. 

Off course one can argue that everything here is INVALID_ARGUMENT, so we do not need the rest. But to me the set seems quite reasonable.

https://github.com/eclipse-kuksa/kuksa-databroker/blob/main/proto/kuksa/val/v2/types.proto#L64

```
message Error {
  ErrorCode code = 1;
  string message = 2;
}

enum ErrorCode {
  OK                = 0;
  INVALID_ARGUMENT  = 1;
  NOT_FOUND         = 2;
  PERMISSION_DENIED = 3;
}
```

Reply:
Given that this could be further refined in the future I think to use ErrorCode if it meets current needs is sufficient for now.